### PR TITLE
Enable external DNS/LB support

### LINKED ISSUE #191: Use external LB and DNS for install
We should have a way to disable haproxy and DNS in bastion when using external LB and DNS services.

### PR DESCRIPTION
**Pull request by cs-zhang**
Fixed: #191 

Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -310,6 +310,16 @@ variable "dns_forwarders" {
...
+variable "lb_ipaddr" {
+    description = "Define the preconfigured external Load Balancer"
+    default = ""    
+}
+
+variable "ext_dns" {
+    description = "Define the preconfigured external DNS and Load Balancer"
+    default = ""
+}
+
```
Comment (bpradipt) [inline variables.tf:322]: update to the helpernode_tag ?

File: docs/var.tfvars-doc.md [hunk 2]
```diff
@@ -134,6 +134,12 @@ The total length of `cluster_id_prefix`.`cluster_id` should not exceed 14 charac
 
 These variables provides miscellaneous customizations. For common usage scenarios these are not required and should be left unchanged.
 
+The following variables are used to set the external DNS and load balancer, use IP address for these two variables. They can be used only when external DNS and load balancer are pre-configured for the OCP installation.
```
Comment (bpradipt) [inline docs/var.tfvars-doc.md]: How about rephrasing it to something like - "The following variables are used to define the IP address for the preconfigured external DNS and the Load-balancer"

Comment (cs-zhang) [reply docs/var.tfvars-doc.md]: Done

File: variables.tf (hunk 1 above)
Comment (cs-zhang) [reply variables.tf:322]: There is no tag on ocp4-helpernode, the PR will be merged in master.

Comment (bpradipt) [reply variables.tf:322]: Was referring to this - https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/cab69442598ed00c63ba141cb6ba107f6600f927/variables.tf#L243 

We should wait for the helpernode PR to be available in main/master branch before committing this.

Comment (Prajyot-Parab) [reply variables.tf:322]: @cs-zhang the changes need to be merged in helpernode first before we can use it here.

Comment (cs-zhang) [reply variables.tf:322]: Christian said the PR will be merged in next week, we can do after it.

**Comment by cs-zhang** [comment]
The PR for ocp4-helpernode has merged into master.

**Comment by yussufsh** [comment]
@cs-zhang Can you please update the helpernode tag?.. we can then merge this.

**Comment by cs-zhang** [comment]
The change for ocp4-helpernode has already in master branch, so no tag is required.

**Comment by yussufsh** [comment]
> The change for ocp4-helpernode has already in master branch, so no tag is required.

The TF module does not use master branch of helpernode directly. You need to update the hash of your commit here https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/fd9751ed8c29a3f8c22114937c3af4f4d9c03065/variables.tf#L266. Same is true for other release-4.x branches.

**Review by yussufsh — APPROVED** [review]
/lgtm
@cs-zhang Hope you have tested the flow with the new helpernode tag. Thanks!

**Review by Prajyot-Parab — APPROVED** [review]
/approve

**Comment by cs-zhang** [comment]
I have tested with the new tag.

**Pull request merged**